pkgs/build-support/dart/pub2nix/package-graph.py

```diff
@@ -0,0 +1,58 @@
+"""
+https://github.com/dart-lang/pub/issues/4522
+This script generates a package_graph.json file.
+"""
+
+import json
+import os
+from pathlib import Path
+from urllib.parse import unquote, urlparse
+
+import yaml
+
+
+def get_package(pubspec_path: Path, dev_dependencies: bool = False):
+    with pubspec_path.open("r", encoding="utf-8") as f:
+        pubspec = yaml.safe_load(f)
+    package = {
+        "name": pubspec["name"],
+        "version": pubspec.get("version") or "0.0.0",
+        "dependencies": list((pubspec.get("dependencies") or {}).keys()),
+    }
+    if dev_dependencies:
+        package["devDependencies"] = list(
+            (pubspec.get("dev_dependencies") or {}).keys()
+        )
+    return package
+
+
+def main() -> None:
+    package_config_file_path = Path(os.environ["packageConfig"])  # noqa: SIM112
+    with package_config_file_path.open("r", encoding="utf-8") as f:
+        package_config = json.load(f)
+    package_graph = []
+    root_package = get_package(Path("pubspec.yaml"), dev_dependencies=True)
+    for data in package_config.get("packages", []):
+        if data["name"] == root_package["name"] or data["rootUri"] == "flutter_gen":
+            continue
+        package_graph.append(
+            get_package(
+                Path(unquote(urlparse(data["rootUri"]).path)).resolve() / "pubspec.yaml"
```

Comment:
```suggestion
                Path(unquote(urlparse(data["rootUri"]).path)) / "pubspec.yaml"
```

If the only use of this path is to open `pubspec.yaml` right away, there's no point to calling `.resolve()` right?

Reply:
To my understanding, this will just translate to something like `/nix/store/5a2q1ibwvyyfpynai4gl94axymzqds48-pub-analyzer_plugin-0.11.3/./pubspec.yaml`, which is a perfectly valid path that Python has no problem opening?